### PR TITLE
Embedded Media

### PR DESCRIPTION
**Pull request by Ashod**
- embedded video: show inside selection marker
- wsd: support embedded media

### CODE REVIEW

File: wsd/DocumentBroker.cpp [hunk 1]
```diff
@@ -3723,6 +3765,66 @@ bool DocumentBroker::isAsyncUploading() const
...
+            uri.setHost(_uriPublic.getHost());
+            uri.setPort(COOLWSD::getClientPortNumber());
+            uri.addQueryParameter("ServerId", Util::getProcessIdentifier());
+            uri.addQueryParameter("Tag", id);
+            uri.addQueryParameter("WOPISrc", wopiSrc);
+
+            std::string mediaUrl;
+            Poco::URI::encode(uri.toString(), "&", mediaUrl); // '&' is reserved in xml/html/svg.
+            object->set("url", mediaUrl);
+            object->set("mimeType", "video/ogg");
```
Comment (eszkadev) [inline wsd/DocumentBroker.cpp:3799]: Here we have `video/ogg` but above in `DocumentBroker::handleMediaRequest` we always put `video/mp4`
Is that correct?

Comment (Ashod) [reply wsd/DocumentBroker.cpp]: Fixed (although I don't think it makes a difference in practice, and anyway we don't know the actual media mime, but best to be consistent, of course).

**Review by eszkadev — APPROVED** [review]
Works as expected

**Comment by eszkadev** [comment]
I've noticed we have to disable "media playback" panel in the sidebar...

**Comment by eszkadev** [comment]
rebased, core part was merged https://gerrit.libreoffice.org/c/core/+/141502

**Comment by vmiklos** [comment]
What is the next step here, do we have to wait till tomorrow so that CI "sees" the new LOK callback from the daily binary?

File: wsd/DocumentBroker.cpp [hunk 2]
```diff
@@ -3729,6 +3771,66 @@ bool DocumentBroker::isAsyncUploading() const
...
+            // Store the original json with the internal, temporary, file URI.
+            _embeddedMedia[id] = json;
+
+            std::string wopiSrc;
+            Poco::URI::encode(Util::split(_uriPublic.toString(), '?').first, ":?#/=&", wopiSrc);
+
+            Poco::URI uri("/cool/media");
+            uri.setScheme(_uriPublic.getScheme());
+            uri.setHost(_uriPublic.getHost());
+            uri.setPort(COOLWSD::getClientPortNumber());
```
[the remark this answers is not in the document]
Comment (eszkadev) [reply wsd/DocumentBroker.cpp:3797]: https://cpci.cbg.collabora.co.uk:8080/job/github_online_master_android_vs_co-22.05/1352/console

**Pull request merged**